blockchain/blocks/src/tipset.rs

```diff
@@ -202,6 +202,24 @@ impl Tipset {
     pub fn tip_epoch(&self) -> &ChainEpoch {
         self.blocks[0].epoch()
     }
+    /// Returns true if tipsets are equal
+    pub fn equals(&self, ots: Tipset) -> bool {
+        if self.blocks().is_empty() && ots.blocks().is_empty() {
+            return true;
+        }
+        if self.blocks.is_empty() || ots.blocks().is_empty() {
+            return false;
+        }
+        if self.blocks().len() != ots.blocks().len() {
+            return false;
+        }
+        for (i, b) in (self.blocks()).iter().enumerate() {
+            if b.cid() != ots.blocks()[i].cid() {
+                return false;
+            }
+        }
+        true
+    }
```

Comment:
Why do you need this over just deriving PartialEq and/or Eq ? Logic seems to be identical and would enable more easy usage and more compact?

Reply:
Actually it's already being deriven right now, but if you need to change the logic of it (maybe ignoring the keys for whatever reason) can just manually implement it, better to be able to use `==` syntax